src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private static final Random RANDOM = new Random();
+
+    public static Figure getRandomFigure(String randomColor) {
+        switch (RANDOM.nextInt(5)) {
+            case 0:
+                return new Square(ColorSupplier.getRandomColor(),
+                        RANDOM.nextDouble() * 10 + 1);
+            case 1:
+                return new Rectangle(ColorSupplier.getRandomColor(),
```

Comment:
`ColorSupplier.getRandomColor()` can be extracted to the variable before the `switch`